src/extensions/cp/ui/PopUpButton.lua

```diff
@@ -157,19 +159,25 @@ end
 ---  * the `Statement`.
 function PopUpButton:doSelectValue(value)
     return If(self.UI)
-    :Then(If(self.menuUI):Is(nil):Then(self:doPress()))
-    :Then(WaitUntil(self.menuUI):TimeoutAfter(TIMEOUT_AFTER))
-    :Then(function(menuUI)
-        for _,item in ipairs(menuUI) do
-            if item:title() == value and item:enabled() then
-                item:doPress()
-                return true
-            end
-        end
-        menuUI:doCancel()
-        return false
-    end)
-    :Then(WaitUntil(self.menuUI):Is(nil):TimeoutAfter(TIMEOUT_AFTER))
+    :Then(
+        If(function() return self.value and self:value() == value end):Then(true)
```

Comment:
I believe that `self.value` will never be `nil` (it's a lazy prop, so it gets created on demand), so this could just be:

Reply:
```suggestion
        If(self.value):Is(value):Then(true)
```